modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/nhttp/HttpCoreNIOListener.java

```diff
@@ -363,6 +361,52 @@ public int compare(InetSocketAddress a1, InetSocketAddress a2) {
             }
         }
     }
+
+    /**
+     * Start specific end points given by InetSockeAddress list
+     *
+     * @param endpointsClosed InetSocketAddresses of endpoints to be started
+     * @throws AxisFault
+     */
+    private void startSpecificEndpoints(List<InetSocketAddress> endpointsClosed) throws AxisFault {
+        Queue<ListenerEndpoint> endpoints = new LinkedList<ListenerEndpoint>();
+
+        // Ensure simple but stable order
+        List<InetSocketAddress> addressList = endpointsClosed;
+        Collections.sort(addressList, new Comparator<InetSocketAddress>() {
+
+            public int compare(InetSocketAddress a1, InetSocketAddress a2) {
+                String s1 = a1.toString();
+                String s2 = a2.toString();
+                return s1.compareTo(s2);
+            }
+
+        });
+
+        for (InetSocketAddress address : addressList) {
+            endpoints.add(ioReactor.listen(address));
+        }
+
+        // Wait for the endpoint to become ready, i.e. for the listener to start accepting
+        // requests.
+        while (!endpoints.isEmpty()) {
+            ListenerEndpoint endpoint = endpoints.remove();
+            try {
+                endpoint.waitFor();
+                if (log.isInfoEnabled()) {
+                    InetSocketAddress address = (InetSocketAddress) endpoint.getAddress();
+                    if (!address.isUnresolved()) {
+                        log.info(name + " started on " + address.getHostName() + ":" + address.getPort());
+                    } else {
+                        log.info(name + " started on " + address);
+                    }
+                }
+            } catch (InterruptedException e) {
+                log.warn("Listener startup was interrupted");
```

Comment:
Is it right to break the code when the thread is interrupted? As it appears, it is a must that we wait till the endpoint is properly initialized before moving forward with the rest of the activities?